packages/amplify-cli/src/index.ts

```diff
@@ -151,18 +152,29 @@ export const run = async (startTime: number): Promise<void> => {
   await displayBannerMessages(input);
   await executeCommand(context);
 
-  const exitCode = process.exitCode || 0;
-  if (exitCode === 0) {
-    context.usageData.emitSuccess();
-  }
-
   // no command supplied defaults to help, give update notification at end of execution
   if (input.command === 'help') {
     // Checks for available update, defaults to a 1 day interval for notification
     notify({ defer: true, isGlobal: true });
   }
 
-  await saveAllEnvParams();
+  if (context.input.command === 'push') {
+    const { providers } = stateManager.getProjectConfig(undefined, { throwIfNotExist: false, default: {} });
+    const CloudFormationProviderName = 'awscloudformation';
```

Comment:
I think if we add that it should be exported from the `amplify-provider-awscloudformation` package itself, but that increases the API surface. I'd like to avoid that unless it's used more than once in the `amplify-cli-internal` package. Thoughts?